baselines/flwr_baselines/scripts/leaf/shakespeare/create_datasets_and_splits.sh

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# Process Shakespeare for non-iid, 20% testset, minimum two lines
+# Delete previous partitions if needed then extract the entire dataset
+echo 'Deleting previous dataset split.'
+cd ${LEAF_ROOT}/data/shakespeare
```

Comment:
I think it would be great to check if the env variable is set and otherwise warn and exit. Maybe something like this?:

```suggestion
if [[ -z "${LEAF_ROOT}" ]]; then
  echo 'The environment variable LEAF_ROOT is not set. Exiting...'
  exit 1
fi

cd ${LEAF_ROOT}/data/shakespeare
```

Reply:
Nice